backend/pong/match_tournament/match_session.py

```diff
@@ -0,0 +1,397 @@
+from typing import Optional, Callable
+from ..game_logic.game_session import GameSession
+from channels.layers import get_channel_layer
+from usermanagement.models import CustomUser, Games
+from asgiref.sync import sync_to_async
+import asyncio
+import logging
+import time
+from enum import Enum, auto
+
+logger = logging.getLogger("match")
+
+DISCONNECT_THRESHOLD = 3
+MATCH_START_TIMEOUT = 10
+RECONNECT_TIMEOUT = 10
+GAME_START_TIMER = 3 # Time in seconds before the game starts
+
+TICK_RATE = 60
+
+WINNING_SCORE = 3
+
+class EndReason(Enum):
+    DISCONNECT_TIMEOUT = auto()
+    DISCONNECTED_TOO_MANY_TIMES = auto()
+    MATCH_START_TIMEOUT = auto()
+    DRAW = auto()
+    SCORE = auto()
+    LOCAL_MATCH_ABORTED = auto()
+
+class MatchSession:
+    def __init__(self, user_id_1: str, user_id_2: Optional[str], on_match_finished: Optional[Callable[[str, str], None]] = None):
+        '''Initialize and start a match between two users'''
+        self._match_id = str(id(self))  # Generating a unique id
+        self._assigned_users = {user_id_1, user_id_2} if user_id_2 is not None else {user_id_1}
```

Comment:
Would it still work if two matches have the same id, but are in different processes?
if not maybe use uuid4:
`
import uuid
uuid.uuid4()
`

Reply:
The match id will be always unique because python guarantees that every object has a unique id. (ID is tied to the memory address of the object)
What do you mean with "different processes"?

Reply:
Django runs on different processes so its possible that two id's are the same. Because they can have the same memory address.

Reply:
Good catch, I didn't know that. I changed it.